File: leet_easy_c++/leet_070/leet_070a.cpp
class Solution {
public:
    int climbStairs(int n) {
        int total = recur_func(n);
        return total;
    }
    int recur_func(int tot){
        if(tot < 0) return 0;
        if(tot == 0) return 1;
        return recur_func(tot - 1) + recur_func(tot - 2);
    }
};
// potential stack overflow. dynamic programming necessary!

File: leet_easy_python/leet_070/leet_070a.py
class Solution(object):
    def climbStairs(self, n):
        if n < 3: return n
        a = 1
        b = 2
        for i in range(2,n):
            tmp = a + b
            a = b
            b = tmp
        return b
        
        """
        :type n: int
        :rtype: int
        """